homeassistant/components/recorder/migration.py

```diff
@@ -2453,7 +2453,7 @@ def __init__(
         self.migration_changes = migration_changes
 
     @abstractmethod
-    def migrate_data(self, instance: Recorder) -> bool:
+    def migrate_data(self, instance: Recorder, /) -> bool:
```

Comment:
Subclass implementations are decorated with `retryable_database_job_method` which extracts the `Recorder` type using `Concatenate`, thus making it positional-only.

```py
Callable[Concatenate[Self, Recorder, P], R]
```